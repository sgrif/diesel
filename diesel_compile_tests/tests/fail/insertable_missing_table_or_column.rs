#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
    }
}

#[derive(Insertable)]
struct Post {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = posts)]
struct Post2 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = users)]
struct User1 {
    name: String,
}

#[derive(Insertable)]
#[diesel(table_name = users)]
struct User2 {
    #[diesel(column_name = name)]
    name: String,
}

fn main() {}
