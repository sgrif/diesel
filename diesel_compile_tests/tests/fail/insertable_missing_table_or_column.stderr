error[E0433]: failed to resolve: use of undeclared crate or module `posts`
  --> $DIR/insertable_missing_table_or_column.rs:11:8
   |
11 | struct Post {
   |        ^^^^ use of undeclared crate or module `posts`

error[E0433]: failed to resolve: use of undeclared crate or module `posts`
  --> $DIR/insertable_missing_table_or_column.rs:16:23
   |
16 | #[diesel(table_name = posts)]
   |                       ^^^^^ use of undeclared crate or module `posts`

error[E0412]: cannot find type `name` in module `users`
  --> $DIR/insertable_missing_table_or_column.rs:24:5
   |
24 |     name: String,
   |     ^^^^ not found in `users`

error[E0425]: cannot find value `name` in module `users`
  --> $DIR/insertable_missing_table_or_column.rs:24:5
   |
24 |     name: String,
   |     ^^^^ not found in `users`

error[E0412]: cannot find type `name` in module `users`
  --> $DIR/insertable_missing_table_or_column.rs:30:28
   |
30 |     #[diesel(column_name = name)]
   |                            ^^^^ not found in `users`

error[E0425]: cannot find value `name` in module `users`
  --> $DIR/insertable_missing_table_or_column.rs:30:28
   |
30 |     #[diesel(column_name = name)]
   |                            ^^^^ not found in `users`
